misc-scripts/canonical_transcripts/select_canonical_transcripts.pl

```diff
@@ -61,183 +65,233 @@
-        
-        if (! defined($old_canonical)) {
+
+        if ( !defined $old_canonical ) {
             # Original canonical transcript is now absent, or never set.
             if ($log_fh) {
                 print $log_fh "//\n";
                 print $log_fh "Old=[undef,undef,undef,undef,undef,undef,undef]\n";
-                printf $log_fh "New=[%s,%s,%s,%s,%s,%s,'%s']\n", @{ $transcript_selector->encode_transcript($new_canonical) };
+                printf $log_fh "New=[%s,%s,%s,%s,%s,%s,'%s']\n",
+                  @{ $transcript_selector->encode_transcript($new_canonical) };
             }
-            push @change_list,[$gene->dbID,$new_canonical->dbID];
+            push @change_list, [ $gene->dbID, $new_canonical->dbID ];
             $canonical_changes++;
-        } elsif ($new_canonical->dbID != $old_canonical->dbID) {
+        }
+
+        elsif ($new_canonical->dbID != $old_canonical->dbID) {
             no warnings 'uninitialized';
-            printf "%s (%s) changed transcript from %s (%s) to %s (%s)\n",
-                $gene->stable_id,$gene->dbID,$old_canonical->stable_id,$old_canonical->dbID,
-                $new_canonical->stable_id,$new_canonical->dbID;
+            printf
+                "%s (%s) changed transcript from %s (%s) to %s (%s)\n",
+                $gene->stable_id,
+                $gene->dbID,
+                $old_canonical->stable_id,
+                $old_canonical->dbID,
+                $new_canonical->stable_id,
+                $new_canonical->dbID;
+
+            push @change_list, [ $gene->dbID, $new_canonical->dbID ];
             $canonical_changes++;
-            
-            push @change_list,[$gene->dbID,$new_canonical->dbID];
-            
+
             if ($verbose) {
                 printf "Old transcript: [%s,%s,%s,%s,%s,%s,%s]\n",
                     @{ $transcript_selector->encode_transcript($old_canonical) };
                 printf "New transcript: [%s,%s,%s,%s,%s,%s,%s]\n",
                     @{ $transcript_selector->encode_transcript($new_canonical) };
             }
-            
+
             if ($log_fh) {
-                print $log_fh "//\n" or die "Unable to log choices. Cannot write to file";
-                printf $log_fh "Old=[%s,%s,%s,%s,%s,%s,'%s']\n", @{ $transcript_selector->encode_transcript($old_canonical) };
-                printf $log_fh "New=[%s,%s,%s,%s,%s,%s,'%s']\n", @{ $transcript_selector->encode_transcript($new_canonical) };
+                print $log_fh "//\n";
+                printf $log_fh "Old=[%s,%s,%s,%s,%s,%s,'%s']\n",
+                  @{ $transcript_selector->encode_transcript($old_canonical) };
+                printf $log_fh "New=[%s,%s,%s,%s,%s,%s,'%s']\n",
+                  @{ $transcript_selector->encode_transcript($new_canonical) };
             }
         }
-        
     }
 }
 
-print "Canonical transcript alterations: ".$canonical_changes." from ".$total_genes." genes\n";
-if ($log_fh) {$log_fh->close;}
```

Comment:
Now fixed on this PR (sorry again):
https://github.com/Ensembl/ensembl/pull/199/files#diff-03f7b81015cecfc399a731e3d81aa4caL198